snyk/models.py

```diff
@@ -164,7 +164,7 @@ def notification_settings(self):
     def invite(self, email: str, admin: bool = False):
         raise SnykNotImplementedError  # pragma: no cover
 
-    def _test(self, path, contents=None):
+    def _test(self, path, contents=None, additional=None):
```

Comment:
How about we add type hints for this method as well, like so:
```python
def _test(self, path: str, contents: typing.IO = None, additional: typing.IO = None)
```

using typing.IO will cover both binary(if necessary) and text file-like objects.

Reply:
This won't quite work as is, as the interface accepts file handles or strings.

Reply:
Thanks for clarifying, API looks ambiguous to me. shouldn't the function calling `_test` be the one reading the file content and passing that on? I guess this is an inherited design smell.

worth a re-factor, though I'm sure we'll have to consider backwards compatibility here.

for instance why `test_pipfile(self, content)` is acting like `test_pipfile_content(self, content) `while what I think would be cleaner is:
```
"""  test_pipfile tests a pip file for issue """
def test_pipfile(self, pip_file: typing.TextIO ) -> bool:
    with open(pip_file) as f:
        self._test(contents=f.read())
```